ddtrace/tracer/option.go

```diff
@@ -1008,3 +1007,20 @@ func WithPropagation() UserMonitoringOption {
 		cfg.PropagateID = true
 	}
 }
+
+func WithAdditionalTransportHeaders(headers map[string]string) StartOption {
+	return func(c *config) {
+		tr, ok := c.transport.(*httpTransport)
+		if ok {
+			for key, value := range headers {
+				tr.headers[key] = value
+			}
+		} else if tr == nil {
+			tr = newHTTPTransport(c.agentURL.String(), c.httpClient)
+			for key, value := range headers {
+				tr.headers[key] = value
+			}
+			c.transport = tr
+		}
+	}
+}
```

Comment:
This is not necessary, `WithHTTPClient` is enough to do what you want to do. You can provide your own transport in your http client and make it add the headers you want on every request. See a more complete answer here https://stackoverflow.com/a/54088988 - I'm mostly referring to option 3.

Reply:
awesome, thanks for the help, i was able to do just this